### ISSUE TITLE
More image loss functions: MS-SSIM

### ISSUE THREAD

**Issue by kateiyas** [pfnet/chainer#2503]
Many image generation methods operate on losses computed directly between images, however e.g. autoencoders perform poorly on pixel-wise methods such as MSE. One recent suggestion is using MS-SSIM loss, which is differentiable and has been shown to perform well-enough for generative models: https://www.cs.colorado.edu/~mozer/Research/Selected%20Publications/reprints/Snelletal2016.pdf

Derivative of MS-SSIM is in this paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4143340/

Somebody partly implemented it in TensorFlow here: http://stackoverflow.com/questions/39051451/ssim-ms-ssim-for-tensorflow

I basically suggest to add this to get one more good measure for similarity between generated images.

**Comment by kateiyas** [comment]
Po-Hsun-Su wrote a pytorch implementation here: https://github.com/Po-Hsun-Su/pytorch-ssim/
